tests/driver_epd_bw_spi/Makefile

```diff
@@ -0,0 +1,10 @@
+BOARD ?= nucleo-f411re
+
+include ../Makefile.tests_common
+
+USEMODULE += xtimer
+USEMODULE += epd_bw_spi
+
+INCLUDES += -I$(APPDIR)
```

Comment:
This is a configuration for your personal setup. I don't think it should be put there (I mean in RIOT). Maybe someone with a nucleo-f411re won't wire the same e-ink paper display than yours and maybe not on the same pins.
So I would simply remove the `epd_bw_spi_params.h` include from the application directory (it should remain generic as much as possible). Note that this would simply forbid the possibility to build this application with a board providing a e-ink paper: the board support won't be able to provide custom defines (different pins, screen size, controller) the usual way.
The x/y display sizes and controller are the same as the default ones, so no need to overwrite them here.

Also note that I would be ok to move the setup specific pin definitions to the default driver params with a comment (use the same one used in the application params.h file, you can even generalize to any Nucleo 64 board I guess, they have a compatible pinout).

Reply:
> This is a configuration for your personal setup. I don't think it should be put there (I mean in RIOT). 

Sure, but it's the only setup I have to test with.

> Maybe someone with a nucleo-f411re won't wire the same e-ink paper display than yours and maybe not on the same pins.

Which is why the pinout is documented...

> So I would simply remove the epd_bw_spi_params.h include from the application directory (it should remain generic as much as possible).

Sure, but that just means that I have to keep it untracked in my code base. Otherwise I won't be able to actually run this application.

> Note that this would simply forbid the possibility to build this application with a board providing a e-ink paper: the board support won't be able to provide custom defines (different pins, screen size, controller) the usual way.

How would removing the header file lead to this application not working with a board having an e-paper display?

> The x/y display sizes and controller are the same as the default ones, so no need to overwrite them here.

I'd much rather have a complete override instead of a partial one in case defaults change.

> Also note that I would be ok to move the setup specific pin definitions to the default driver params with a comment (use the same one used in the application params.h file, you can even generalize to any Nucleo 64 board I guess, they have a compatible pinout).

Sorry, but I don't think it makes any sense: my personal setup can't be included explicitly, but it's fine to do it implicitly as driver defaults?

Reply:
> How would removing the header file lead to this application not working with a board having an e-paper display?

No, I meant that keeping the header file you added would forbid this: it will fail with the CI (because of redefined defines).

> I don't think it makes any sense: my personal setup can't be included explicitly, but it's fine to do it implicitly as driver defaults?

I said for convenience with your setup. Since you are contributing this driver, then it's ok to have these defaults. And other users (or boards support) are still allowed to provide their own setup. In the current state of this PR, this is not possible without modifying RIOT (which is annoying).

Reply:
> No, I meant that keeping the header file you added would forbid this: it will fail with the CI (because of redefined defines).

So what would be the proper way to have these overrides?

Reply:
You could use CFLAGS from the command line. But I would avoid any custom setup specific configuration in a RIOT application, this is not generic and cannot be reused in another setup.
If you set the driver default params with your preferred values (let's say it's a privilege given to the initial contributor), then you won't have to change anything. But if someone wants to reuse this application for its needs with a different setup, he can add CFLAGS to the build command line without having to change any line of code. If it's a custom setup hardwired on a board (like the stm32l0 disco), then the defines will have to defined in board.h of this board (board.h must be included in `xx_params.h` header and the application can be built as is.